selene_sdk/train_model.py

```diff
@@ -242,60 +247,84 @@ def __init__(self,
+            self.init_test()
         if checkpoint_resume is not None:
-            checkpoint = torch.load(
-                checkpoint_resume,
-                map_location=lambda storage, location: storage)
-            if "state_dict" not in checkpoint:
-                raise ValueError("Selene does not support continued "
-                    "training of models that were not originally "
-                    "trained using Selene.")
-
-            self.model = load_model_from_state_dict(
-                checkpoint["state_dict"], self.model)
-
-            self._start_step = checkpoint["step"]
-            if self._start_step >= self.max_steps:
-                self.max_steps += self._start_step
-
-            self._min_loss = checkpoint["min_loss"]
-            self.optimizer.load_state_dict(
-                checkpoint["optimizer"])
-            if self.use_cuda:
-                for state in self.optimizer.state.values():
-                    for k, v in state.items():
-                        if isinstance(v, torch.Tensor):
-                            state[k] = v.cuda()
-
-            logger.info(
-                ("Resuming from checkpoint: step {0}, min loss {1}").format(
-                    self._start_step, self._min_loss))
-
+            self.load_checkpoint(checkpoint_resume)
+
+    def load_checkpoint(self, checkpoint_resume):
+        checkpoint = torch.load(
+            checkpoint_resume,
+            map_location=lambda storage, location: storage)
+        if "state_dict" not in checkpoint:
+            raise ValueError("Selene does not support continued "
+                "training of models that were not originally "
+                "trained using Selene.")
+
+        self.model = load_model_from_state_dict(
+            checkpoint["state_dict"], self.model)
+
+        self._start_step = checkpoint["step"]
+        if self._start_step >= self.max_steps:
+            self.max_steps += self._start_step
+
+        self._min_loss = checkpoint["min_loss"]
+        self.optimizer.load_state_dict(
+            checkpoint["optimizer"])
+        if self.use_cuda:
+            for state in self.optimizer.state.values():
+                for k, v in state.items():
+                    if isinstance(v, torch.Tensor):
+                        state[k] = v.cuda()
+
+        logger.info(
+            ("Resuming from checkpoint: step {0}, min loss {1}").format(
+                self._start_step, self._min_loss))
+                
+    def init_train(self):
+        self._start_step = 0
         self._train_logger = _metrics_logger(
                 "{0}.train".format(__name__), self.output_dir)
+        self._train_logger.info("loss")
+        if self.use_scheduler:
+            self.scheduler = ReduceLROnPlateau(
+                self.optimizer,
+                'min',
+                patience=16,
+                verbose=True,
+                factor=0.8)
```

Comment:
In the docs for `use_scheduler`, can we include the parameters `patience=16` and `factor=0.8` for users to know exactly how the LR will be reduced?

Reply:
or maybe we can defer specifying scheduler to the model file since the  user may want to change these parameters or even use a different scheduler. Given that optimizer is specified in the model file already, I guess it makes sense to put scheduler there too.  

I still like the idea of specifying the optimizer (and maybe scheduler) through yaml file haha - but scheduler need to take the optimizer as an argument, I am not sure whether this can be done through our parser.

Reply:
do you have an idea you want to try re:adding the scheduler through YAML? if it's not easy/fast to do then we won't try to add it

Reply:
No specifying a scheduler seems to be beyond what YAML can do at this point? agree we can ignore this for now